File: src/dataframe/rust/stats/statistical_tests/chi_square.test.R
#!/usr/bin/env Rscript

# R script for Chi-square test comparisons
# Usage: Rscript chi_square.test.R <function_name> <args...>

args <- commandArgs(trailingOnly = TRUE)

if (length(args) < 1) {
  stop("Function name required")
}

function_name <- args[1]

if (function_name == "chisq.test.independence") {
  # Chi-square test of independence
  contingency_table <- jsonlite::fromJSON(args[2])
  
  # Convert to matrix (fromJSON already gives us a matrix for 2D arrays)
  matrix_data <- as.matrix(contingency_table)
  
  # Perform chi-square test
  result <- chisq.test(matrix_data)
  
  cat(jsonlite::toJSON(list(
    test_statistic = result$statistic,
    p_value = result$p.value,
    reject_null = result$p.value < 0.05
  ), auto_unbox = TRUE))
  
} else {
  stop(paste("Unknown function:", function_name))
}

File: tests/statistical_tests/spot_checks/chi-square.test.ts
import { stats } from "../../../src/dataframe/mod.ts";

Deno.test("Chi-Square Test Spot Check", () => {
  console.log("\n📊 Chi-Square Test of Independence");
  console.log("-".repeat(80));

  const contingencyTable = [
    [10, 20, 30],
    [15, 25, 35],
  ];

  const result = stats.test.categorical.chiSquare({
    contingencyTable,
  });

  console.log("Contingency Table:", contingencyTable);
  console.log("Chi-squared statistic:", result.test_statistic.value);
  console.log("df:", result.degrees_of_freedom);
  console.log("p-value:", result.p_value);
});

console.log("\n" + "=".repeat(80));
console.log("CHI-SQUARE TEST SPOT CHECK");
console.log("=".repeat(80));
